pkg/redislocker/redislocker.go

```diff
@@ -0,0 +1,148 @@
+package redislocker
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"golang.org/x/exp/slog"
+
+	"github.com/tus/tusd/v2/pkg/handler"
+)
+
+var (
+	LockExchangeChannel = "tusd_lock_release_request_%s"
+	LockReleaseChannel  = "tusd_lock_released_%s"
+	LockExpiry          = 8 * time.Second
+)
+
+type LockExchange interface {
```

Comment:
`LockExchange` seems to be only used in `BidirectionalLockExchange`, so these interface could be merged.